### ISSUE TITLE
Incorrect warning "This server's certificate will be distrusted by Google and Mozilla from September 2018"

### ISSUE THREAD

**Issue by manuelhartl** [ssllabs/ssllabs-scan#638]
Hi,

The website reports "This server's certificate will be distrusted by Google and Mozilla from September 2018" for www.ipg-online.com:
https://www.ssllabs.com/ssltest/analyze.html?d=www.ipg-online.com

This is wrong. the TLS server certificate is issued by Digicert!

Kind regards,
Manuel.

**Comment by anand-bhat** [comment]
Also see https://github.com/ssllabs/ssllabs-scan/issues/632

**Comment by roycewilliams** [comment]
The fourth path listed in the results has Verisign as its CA.

**Comment by manuelhartl** [comment]
Yes. but this isn’t a problem. According to Google/Mozilla they will only complain about TLS *Server* certificates issued by the symantec/verisign CA. not about intermediates or roots!

**Comment by manuelhartl** [comment]
Hi Anand. Right. Same complaint, same reasoning. The browser warnings are only about END-END-Certificates (TLS Server certificates). Not about the Roots/Intermediates.

**Comment by roycewilliams** [comment]
From [the consensus proposal](https://groups.google.com/a/chromium.org/forum/#!topic/blink-dev/eUAKwjihhBs%5B251-275%5D):

> Chrome 70 will distrust TLS certificates issued from Symantec’s old infrastructure:
> 
> In order to complete this migration, we will be removing trust in all certificates issued by Symantec’s old infrastructure in Chrome 70. This includes any replacement certificates issued by Symantec prior to the transition to the non-Symantec-operated “Managed Partner Infrastructure”. Chrome 70 is tentatively scheduled to first reach Beta on September 13, 2018 and Stable on October 23, 2018, which is approximately 5 months after Chrome 66’s corresponding dates.
> 
> By these dates, affected site operators will need to have fully replaced any TLS server certificates **issued from Symantec’s old infrastructure**, using any trusted CA including the new Managed Partner Infrastructure. Failure to migrate a site to one of these two options will result in breakage when Chrome 70 is released.

My interpretation is that the entire chain is within scope, not just the end certificates.

**Comment by anand-bhat** [comment]
If this is the case, Chrome's warning in the console is incorrect. It shows warnings for sites such as affiliate-program.amazon.com that are using a soon to be distrusted leaf certificate.

> The SSL certificate used to load resources from https://affiliate-program.amazon.com will be distrusted in M70. Once distrusted, users will be prevented from loading these resources. See https://g.co/chrome/symantecpkicerts for more information.

However, Chrome does not display any such warnings for https://www.ipg-online.com or https://api-s.paypal.com.

**Comment by manuelhartl** [comment]
HI,

The key term here is “TLS Server certificates”: TLS server certificates issued from Symantec’s old infrastructure, 
They are not taking about other certificates than TLS server certificates!

Also as Anand pointed out - the Chrome Betas implementation do exactly what I describe.

Kind regards,
Manuel.

**Comment by jtdowney** [comment]
If you [read the Chrome source code](https://chromium-review.googlesource.com/c/chromium/src/+/830767) it is only distrusting certificate chains where the leaf was issued by a Symantec cert. If the certificate was issued by DigiCert but the chain includes Symantec there isn't an issue. This is particularly the case if certificate pinning had pinned the `VeriSign Class 3 Public Primary Certification Authority - G5` root. The cross-sign of `DigiCert Global Root G2` by `VeriSign Class 3 Public Primary Certification Authority - G5` would allow these pins to still function and for Chrome to not distrust the certificate.

You can see this case on [api.braintreegateway.com](https://www.ssllabs.com/ssltest/analyze.html?d=api.braintreegateway.com). SSL Labs reports that it will be distrusted but there is no warning in Chrome stable and Chrome Canary 70 does not distrust the certificate.

For Mozilla, they're removing the Symantec roots from their trust store shipped with Firefox. So when Firefox goes to check the chain it knows about `DigiCert Global Root G2` and trusts the certificate.

You can see this already on Firefox Beta by setting `security.pki.distrust_ca_policy` to `2`.

**Comment by naumanshah03** [comment]
Hi all,

@manuelhartl Please check the changes on dev.ssllabs.com
https://dev.ssllabs.com/ssltest/analyze.html?d=www.ipg-online.com

@anand-bhat thank you
https://dev.ssllabs.com/ssltest/analyze.html?d=affiliate-program.amazon.com&hideResults=on

@roycewilliams thank you. Trust path built is checked. You read the implementation given by Google below

@jtdowney thank you
https://dev.ssllabs.com/ssltest/analyze.html?d=api.braintreegateway.com&hideResults=on

There were addition done by Google which is stated on this page - https://chromium.googlesource.com/chromium/src/+/master/net/data/ssl/symantec/#implementation-details

Now, any certificate which leads to a Symantec certificate but contains certificate which has an intermediate CA certificate containing the excluded sub/managed CA SPKI in trust path, such domains will not be affected by Symantec distrust.

For SSL Labs 1.32.4 will have this latest implementation.

Thank you all.

**Comment by roycewilliams** [comment]
Nice. And thanks to everyone for being persistent and polite. :)

**Comment by manuelhartl** [comment]
Hi,

The scan is now as expected on https://dev.ssllabs.com/ssltest/analyze.html?d=www.ipg-online.com <https://dev.ssllabs.com/ssltest/analyze.html?d=www.ipg-online.com>

i also did regression testing for www4.ipg-online.com <http://www4.ipg-online.com/>. Here the warning still is displayed (which is correct)

Thanks!
Manuel.

**Comment by halisa88** [comment]
It appears that https://dev.ssllabs.com/ssltest is showing accurate information, when will this be applied to prod so that rotated certificates are not showing up as an error for mitigated certs.

**Comment by roycewilliams** [comment]
Should these have been fixed as part of this fix? They are currently trusted by Chrome 69 and Firefox 62, but report as untrusted:

https://www.ssllabs.com/ssltest/analyze.html?d=www.nanuq-alaska.net
https://www.ssllabs.com/ssltest/analyze.html?d=www.astac.net
https://www.ssllabs.com/ssltest/analyze.html?d=www.akcache.org

The main issuers listed for the above (and others) are:

GeoTrust DV SSL CA - G3
RapidSSL SHA256 CA

If this is a different issue, I'll open a separate issue.

**Comment by jtdowney** [comment]
@roycewilliams The distrust for Chrome is in 70, which I am running and those sites are not trusted with `NET::ERR_CERT_SYMANTEC_LEGACY`. These certificates appear to have not been reissued under the new DigiCert issuing platform. This GitHub issue was regarding certificates that were re-issued but still contained a cross-signed root for clients that pinned the root.

**Comment by roycewilliams** [comment]
@jtdowney Ah, gotcha - thanks!

**Comment by roycewilliams** [comment]
@jtdowney Interestingly, I just loaded all three of these with the latest Chrome 70 beta available at this writing (specifically, Version 70.0.3538.22 (Official Build) beta (64-bit)) ... and none of them are untrusted by that Chrome.

**Comment by jtdowney** [comment]
Still distrusted on Canary, not sure why it isn't done on Beta.

![image](https://user-images.githubusercontent.com/44654/45882461-d8bdb480-bd73-11e8-94f5-2d2b2e0647b5.png)

**Comment by roycewilliams** [comment]
Here's another one:

https://www.ssllabs.com/ssltest/analyze.html?d=owa.gci.com&hideResults=on

Regardless of whether or not Canary (or even the beta) is throwing errors on these, the fact that current Chrome and current Firefox don't complain *at all*, alongside the fact that Labs declared that Chrome and Firefox "are not trusted", means that something is broken.

**Comment by jtdowney** [comment]
> current Chrome and current Firefox don't complain at all

I disagree with that, the developer console in Chrome says:

> The SSL certificate used to load resources from https://owa.gci.com will be distrusted in M70. Once distrusted, users will be prevented from loading these resources. See https://g.co/chrome/symantecpkicerts for more information.

The SSL Labs tool is correct; these certificates are distrusted, it just hasn't hit stable yet. However, it will very soon.

Besides that, you hijacked an actual issue that the tool had, which was a more complicated case involving cross-signed roots. These sites you keep bringing up are not that case.

**Comment by roycewilliams** [comment]
I'm speaking from the perspective of IT non-specialists, management, and general admins, for whom this disconnect will indeed be nonintuitive, confusing, and likely to cause a number of diagnostic false starts - (and for whom the Developer Console may _eventually_ resolve the disconnect if they dig around for a while - but they shouldn't, IMO, have to do that).

I think we're arguing semantics here. Google and Firefox _as organizations_ clearly distrust these certs, and have been moving for a long time for Chrome and Firefox _the browsers_ to actively distrust them. Most admins and other "end users" of the SSL Labs Server test, in my opinion, are likely to believe that the phrase "This server's certificate is distrusted by Google and Mozilla" means _their respective browsers_, not _the organizations_ (and yes, I'm fully aware that "Google" != "Chrome"). 

Even the earlier phrase "This server's certificate will be distrusted by **Google** and Mozilla from September 2018" is insufficiently precise, because clearly Google and Mozilla _the organizations_ distrusted these certs well before then, and that it was _the browsers_ that were going to start complaining? Are we talking about _organizational_ distrust, or _browser_ distrust? There's simply no need for the ambiguity.

That being said, you're right that I (inadvertently, by attrition) hijacked the thread. I originally brought up what I thought were related mismatches, and got bogged down in the UX problem. I'd happily start a new thread if I thought that it might make a difference.

**Comment by roycewilliams** [comment]
The lack of distrust in beta must have been transient - 70.0.3538.35 now distrusts SYMANTEC_LEGACY certs, as expected.
